File: C/bogosort/gemma2-9b-it/gemma2-9b-it.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

void bogoSort(int arr[], int n) {
    srand(time(NULL));
    while (1) {
        for (int i = 0; i < n - 1; i++) {
            if (arr[i] > arr[i + 1]) {
                break;
            }
        }
        if (i == n - 1) {
            break;
        }
        for (int i = 0; i < n; i++) {
            int j = rand() % n;
            int temp = arr[i];
            arr[i] = arr[j];
            arr[j] = temp;
        }
    }
}

int main() {
    int arr[10];
    for (int i = 0; i <= 10; i++) {
        arr[i] = rand() % 1000;
    }

    bogoSort(arr, 10);

    return 0;
}

File: Java/bogosort/gemma2-9b-it/Gemma2_9b_it.java
import java.util.Arrays;
import java.util.Random;
public class Gemma2_9b_it {

    public static void bogoSort(int[] arr) {
        while (!isSorted(arr)) {
            shuffle(arr);
        }
    }

    private static boolean isSorted(int[] arr) {
        for (int i = 1; i < arr.length; i++) {
            if (arr[i] < arr[i - 1]) {
                return false;
            }
        }
        return true;
    }

    private static void shuffle(int[] arr) {
        for (int i = arr.length - 1; i > 0; i--) {
            int j = (int) (Math.random() * (i + 1));
            int temp = arr[i];
            arr[i] = arr[j];
            arr[j] = temp;
        }
    }

    public static void main(String[] args) {
        Random rand = new Random();

        // Gerar 10 valores aleatórios
        int[] arr = new int[10];
        for (int i = 0; i < arr.length; i++) {
            arr[i] = rand.nextInt(10000);  // valores aleatórios entre 0 e 9999
        }

        // Chamar o algoritmo bogoSort para ordenar o array
        bogoSort(arr);
    }
}